stixcore/idb/manager.py

```diff
@@ -45,15 +60,160 @@ def data_root(self, value):
+        try:
+            with open(self._data_root / IDB_VERSION_HISTORY_FILE) as f:
+                self.history = json.load(f)
+        except EnvironmentError:
+            raise ValueError(f'No IDB version history found at: {self._data_root / IDB_VERSION_HISTORY_FILE}')
+
+    def find_version(self, utc=None):
+        """searches the local version history for a IDB operational at a given time
+
+        Parameters
+        ----------
+        utc : `datetime`, optional
+            the time point of the IDB operation, by default None
+
+        Returns
+        -------
+        `str`
+            a version label
+        """
+        if not utc:
+            try:
+                return self.history[0]['version']
+            except Exception as e:
+                logger.error(str(e))
+            return ''
+        for item in self.history:
+            if utc > dtparser.parse(item['validityPeriod'][0]) and utc<=dtparser.parse(item['validityPeriod'][1]):
+                return item['version']
+
+        logger.error(f"No IDB version found for Time: {utc}")
+        return ''
+    #TODO replace default URL with a public FHNW server
+    def download_version(self, version_label, force=False, url="https://nicky.thecrag.com/public/stix/"):
+        """Downloads and installs an IDB version of a public available URL.
+
+        Parameters
+        ----------
+        version_label : `str` or (`int`, `int`, `int`)
+            a version definition
+        force : `bool`, optional
+            set to True to override the local version, by default False
+        url : `str`, optional
+            public available IDB versions folder, by default "https://nicky.thecrag.com/public/stix/"
+
+        Returns
+        -------
+        `bool`
+            was the download and installation successfully
+
+        Raises
+        ------
+        ValueError
+        """
+        if self.has_version(version_label) and force == False:
+            raise ValueError(f'IDB version {version_label} already available locally. Use force=True if you would like to override')
+
+        if force == True:
+            try:
+                Path(self._get_filename_for_version(version_label)).unlink()
+            except Exception as e:
+                logger.warn(e)
+
+        vlabel = (IDB_VERSION_PREFIX + IdbManager.get_version_label(version_label))
+        vdir = self.data_root / vlabel
+        try:
+            if not vdir.exists():
+                os.mkdir(vdir)
+            u = f"Download idb from git: {url + vlabel + '.zip'}"
+            urllib.request.urlretrieve(url + vlabel + ".zip", vdir / "idb.zip")
+
+            with zipfile.ZipFile(vdir / "idb.zip", 'r') as zip_ref:
+                zip_ref.extractall(vdir / "raw")
+
+            IdbManager.convert_mib_2_sqlite(vdir / "raw" / ("STIX-IDB-"+IdbManager.get_version_label(version_label)) / "idb", self._get_filename_for_version(version_label), IdbManager.get_version_label(version_label))
```

Comment:
This is a very long line maybe assign to variables and pass the variables?

Reply:
are line brakes within function call ok?